tests/manage/pv_services/test_resource_deletion_during_pod_pvc_deletion.py

```diff
@@ -0,0 +1,320 @@
+
+        # Do setup for running IO on pods
+        log.info("Setting up pods for running IO")
+        for pod_obj in self.pod_objs:
+            pod_obj.workload_setup(storage_type='fs')
+        log.info("Setup for running IO is completed on pods")
+
+        # Start IO on each pod
+        log.info("Starting IO on pods")
+        for pod_obj in self.pod_objs:
+            pod_obj.run_io(storage_type='fs', size=f'{self.pvc_size_int - 1}G')
+        log.info("IO started on all pods.")
+
+        # Start deleting pods
+        pod_bulk_delete = executor.submit(self.delete_pods, self.pod_objs)
+
+        if operation_to_disrupt == 'delete_pods':
+            ret = self.verify_resource_deletion(
+                get_all_pods, initial_num_of_pods
+            )
+            assert ret, "Wait timeout: Pods are not being deleted."
+            logging.info(
+                f"Pods deletion has started."
+            )
+            disruption.delete_resource()
+
+        pods_deleted = pod_bulk_delete.result()
+
+        assert pods_deleted, "Deletion of pods failed."
+
+        # Verify pods are deleted
+        for pod_obj in self.pod_objs:
+            assert pod_obj.ocp.wait_for_delete(pod_obj.name), (
+                f"Pod {pod_obj.name} is not deleted"
+            )
+        logging.info("Verified: Pods are deleted.")
+        self.pod_objs.clear()
+
+        # Verify that the mount point is removed from nodes after deleting pod
+        for node, pvs in node_pv_dict.items():
+            cmd = f'oc debug nodes/{node} -- df'
+            df_on_node = run_cmd(cmd)
+            for pv in pvs:
+                assert pv not in df_on_node, (
+                    f"{pv} is still present on node {node} after "
+                    f"deleting the pods."
+                )
+
+        # Start deleting PVCs
+        pvc_bulk_delete = executor.submit(delete_pvcs, self.pvc_objs)
+
+        if operation_to_disrupt == 'delete_pvcs':
+            ret = self.verify_resource_deletion(
+                get_all_pvcs, initial_num_of_pvc
+            )
+            assert ret, "Wait timeout: PVCs are not being deleted."
+            logging.info(
+                f"PVCs deletion has started."
+            )
+            disruption.delete_resource()
+
+        pvcs_deleted = pvc_bulk_delete.result()
+
+        assert pvcs_deleted, "Deletion of PVCs failed."
+
+        # Verify PVCs are deleted
+        for pvc_obj in self.pvc_objs:
+            assert pvc_obj.ocp.wait_for_delete(pvc_obj.name), (
+                f"PVC {pvc_obj.name} is not deleted"
+            )
+        logging.info("Verified: PVCs are deleted.")
+        self.pvc_objs.clear()
+
+        # Verify PVs are deleted
+        for pv_obj in pv_objs:
+            assert pv_obj.ocp.wait_for_delete(pv_obj.name), (
+                f"PV {pv_obj.name} is not deleted"
+            )
+        logging.info("Verified: PVs are deleted.")
+
+        # TODO: Verify PV using ceph toolbox. PV should be deleted.
```

Comment:
the bug is ON_QA.
We can now do this validation.
See https://github.com/red-hat-storage/ocs-ci/issues/502

Reply:
Yes, that is now available. Thanks Elad. 
I will update this using 'verify_pv_exists_in_backend' method in PVC

Reply:
Added function to verify volume is deleted in backend.
'verify_pv_exists_in_backend' method in PVC class is not used because it can only be used to verify volume is present.